src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpUtilities.cs

```diff
@@ -531,5 +532,25 @@ private static bool IsHex(char ch)
                 // Check if less than 6 representing chars 'a' - 'f'
                 || (uint)((ch | 32) - 'a') < 6u;
         }
+
+        public static AltSvcHeader? GetEndpointAltSvc(System.Net.IPEndPoint endpoint, HttpProtocols protocols)
+        {
+            var hasHttp1OrHttp2 = protocols.HasFlag(HttpProtocols.Http1) || protocols.HasFlag(HttpProtocols.Http2);
+            var hasHttp3 = protocols.HasFlag(HttpProtocols.Http3);
+
+            if (hasHttp1OrHttp2 && hasHttp3)
+            {
+                // 86400 is a cache of 24 hours.
+                // This is the default cache if none is specified, but it appears that all popular HTTP/3
+                // websites explicitly specifies a cache duration in the header.
+                var text = "h3=\":" + endpoint.Port.ToString(CultureInfo.InvariantCulture) + "\"; ma=86400";
+                var bytes = Encoding.ASCII.GetBytes($"\r\nAlt-Svc: h3=\":" + endpoint.Port.ToString(CultureInfo.InvariantCulture) + "\"; ma=86400");
```

Comment:
Ah, I hadn't noticed you need to return the string and bytes.
```suggestion
                var bytes = Encoding.ASCII.GetBytes("\r\nAlt-Svc: " + text);
```